tests/reportArticleTest.js

```diff
@@ -0,0 +1,173 @@
+  title: 'Enough is Enough!',
+  description: 'This is a call for Revolt',
+  body: 'My people the time has come to revolt against this new government',
+  image: 'null'
+};
+
+let userToken;
+let adminToken;
+let slug;
+
+describe('Report Articles', () => {
+  before((done) => {
+    chai
+      .request(app)
+      .post('/api/auth/signup')
+      .send(userDetail)
+      .end((err, res) => {
+        const { token } = res.body;
+        userToken = token;
+      });
+    chai
+      .request(app)
+      .post('/api/auth/signup')
+      .send(adminDetail)
+      .end((err, res) => {
+        const { token } = res.body;
+        adminToken = token;
+        done();
+      });
+  });
+
+  after((done) => {
+    resetDB();
+
+    done();
+  });
+
+  describe('Report an article', () => {
+    after((done) => {
+      resetDB();
+
+      done();
+    });
+
+    it('Returns the right response when an article is created', (done) => {
+      chai
+        .request(app)
+        .post('/api/articles')
+        .set({ authorization: userToken, Accept: 'application/json' })
+        .send(article)
+        .end((err, res) => {
+          expect(res.status).to.equal(201);
+          const { title } = res.body.article;
+          expect(title).to.equal('Enough is Enough!');
+          expect(res.body.message).to.equal('You have created an article successfully');
+          const articleSlug = res.body.article.slug;
+          slug = articleSlug;
+          done();
+        });
+    });
+
+    it('Returns 401 error when user is not authenticated', (done) => {
+      chai
+        .request(app)
+        .post(`/api/articles/${slug}/report`)
+        .set({ authorization: 'userToken', Accept: 'application/json' })
+        .send({ violation: 'plagiarisism' })
+        .end((err, res) => {
+          expect(res.status).to.equal(401);
+          expect(res.body.message).to.equal('jwt malformed');
+          done();
+        });
+    });
+
+    it('Returns 404 error when article can not be found', (done) => {
+      chai
+        .request(app)
+        .post('/api/articles/slug/report')
+        .set({ authorization: userToken, Accept: 'application/json' })
+        .send({ violation: 'plagiarisism' })
+        .end((err, res) => {
+          expect(res.status).to.equal(404);
+          expect(res.body.error).to.equal('Article Not found!');
+          done();
+        });
+    });
+
+    it('Should report an article successfully', (done) => {
+      chai
+        .request(app)
+        .post(`/api/articles/${slug}/report`)
+        .set({ authorization: userToken, Accept: 'application/json' })
+        .send({ violation: 'plagiarisism' })
+        .send(article)
+        .end((err, res) => {
+          expect(res.status).to.equal(201);
+          expect(res.body.message).to.equal('You have reported this article successfully');
+          done();
+        });
+    });
+
+    it('Should report an article successfully when field is not field', (done) => {
```

Comment:
Nice job with the tests, But this test and the one above look exactly the same but with different names.

Reply:
Thanks for pointing out this, I'll fix it now